src/test/java/rx/BackpressureTests.java

```diff
@@ -424,25 +428,69 @@ public void testOnBackpressureDrop() {
             .map(SLOW_PASS_THRU).take(NUM).subscribe(ts);
             ts.awaitTerminalEvent();
             ts.assertNoErrors();
-            
-            
+
             List<Integer> onNextEvents = ts.getOnNextEvents();
             assertEquals(NUM, onNextEvents.size());
 
             Integer lastEvent = onNextEvents.get(NUM - 1);
-            
+
             System.out.println("testOnBackpressureDrop => Received: " + onNextEvents.size() + "  Emitted: " + c.get() + " Last value: " + lastEvent);
             // it drop, so we should get some number far higher than what would have sequentially incremented
             assertTrue(NUM - 1 <= lastEvent.intValue());
         }
     }
+
+    @Test(timeout = 10000)
+    public void testOnBackpressureDropWithCallback() {
+        for (int i = 0; i < 100; i++) {
+            final AtomicInteger emitCount = new AtomicInteger();
+            final AtomicInteger dropCount = new AtomicInteger();
+            final AtomicInteger passCount = new AtomicInteger();
+            final int NUM = (int) (RxRingBuffer.SIZE * 1.5); // > 1 so that take doesn't prevent buffer overflow
+            TestSubscriber<Integer> ts = new TestSubscriber<Integer>();
+            firehose(emitCount).onBackpressureDrop(new Action1<Integer>() {
+                @Override
+                public void call(Integer i) {
+                    dropCount.incrementAndGet();
+                }
+            })
+            .doOnNext(new Action1<Integer>() {
+                @Override
+                public void call(Integer integer) {
+                    passCount.incrementAndGet();
+                }
+            })
+            .observeOn(Schedulers.computation())
+            .map(SLOW_PASS_THRU).take(NUM).subscribe(ts);
+            ts.awaitTerminalEvent();
+            ts.assertNoErrors();
+            
+            List<Integer> onNextEvents = ts.getOnNextEvents();
+            Integer lastEvent = onNextEvents.get(NUM - 1);
+            System.out.println(testName.getMethodName() + " => Received: " + onNextEvents.size() + " Passed: " + passCount.get() + " Dropped: " + dropCount.get() + "  Emitted: " + emitCount.get() + " Last value: " + lastEvent);
+            assertEquals(NUM, onNextEvents.size());
+            // in reality, NUM < passCount
+            assertTrue(NUM <= passCount.get());
+            // it drop, so we should get some number far higher than what would have sequentially incremented
+            assertTrue(NUM - 1 <= lastEvent.intValue());
+            assertTrue(0 < dropCount.get());
+            assertEquals(emitCount.get(), passCount.get() + dropCount.get());
+        }
+    }
+
     @Test(timeout = 10000)
     public void testOnBackpressureDropSynchronous() {
         for (int i = 0; i < 100; i++) {
+            final AtomicInteger dropCount = new AtomicInteger();
```

Comment:
These changes should be in a separate test method. Leave the ```testOnBackpressureDropSynchronous``` as is and introduce ```testOnBackpressureDropWithActionSynchronous```.